rllib/core/rl_module/marl_module.py

```diff
@@ -242,6 +242,22 @@ def __getitem__(self, module_id: ModuleID) -> RLModule:
         self._check_module_exists(module_id)
         return self._rl_modules[module_id]
 
+    def __setitem__(self, module_id: ModuleID, module: RLModule) -> None:
+        """Modifies an existing module and assign it to the new module object.
+
+        Args:
+            module_id: The module ID to add.
+            module: The module to add.
+        """
+        try:
+            self._check_module_exists(module_id)
+        except ValueError:
+            raise ValueError(
+                f"Module ID {module_id} does not exist. Use add_module() to add a "
```

Comment:
this is a bit counter-intuitive. just a suggestion.
dict[key] usually works for new key as well.

Reply:
With Jun here. This seems likely to cause confusion, also for us developers.
Like: Why would you allow re-setting an existing module key to some other (new?) module? Would the new module not also require the logic currently handled by `add_module()` to be applied to it?

Reply:
good question. It is used for ddpifying sub-modules and maybe un-ddpifying them if needed. I think setting a new module from scratch should not be allowed via `__setitem__()` since `add_module()` is the entry point for that. The consolidation that I propose here to avoid confusion, is to remove `__setitem__` entry point, and use `add_module()` with the `override=True`. So instead of this:
```
module[key] = DDP(module[key])
```
do
```
module.add_module(key, DDP(module[key]), override=True)
```

Reply:
The other solution is to remove `add_module` api and use `__setitem__` instead. So adding would be 
`module[key] = new_module` and overriding it would also be `module[key] = DDP(module[key])`

Reply:
Let me know what you think about either of the solutions so I can go with one or the other. @gjoliver @sven1977

Reply:
Yes, I like the forced explicitness of add_module(override=True). Also, please throw an error, if override=True and the module key doesn't exist .. just in case :)

Reply:
I do prefer `add_module` as users might not even know which dict/property inside rl trainer to add a new module for.

Reply:
yes, I do raise an error if override is true and the key exists already. I'll take that as the majority vote :)